### DISCUSSION TITLE
What will the microservice architecture be like in Hilla?

### DISCUSSION THREAD

**Discussion by naseef0** [vaadin/hilla#1101] [General]
I need to know What will the microservice architecture be like in Hilla?

**Comment by tarekoraby** [comment]
Hilla only requires a thin endpoint layer to communicate with the frontend. The rest of the backend can continue to employ a microservice setup that Hilla endpoints would communicate with.

**Reply by naseef0** [reply]
@tarekoraby thanks

**Reply by rakeshkesavan** [reply]
@tarekoraby is there any example repository for reference.
Thanks

**Reply by tarekoraby** [reply]
I'm not aware of any example repo.

**Reply by tarekoraby** [reply]
@naseef0 , @rakeshkesavan, here is one example by @marcushellberg: https://hilla.dev/blog/using-microservices-from-hilla/.

**Reply by naseef0** [reply]
@tarekoraby Thanks :)

**Reply by muhammadismailkhan0009** [reply]
@tarekoraby i saw the repo example. a question: won't it incur latency in data fetch cuz we are making 2 http requests to fetch data?
for example, 1st request goes from front-end to hilla app, and 2nd one goes from hilla app to rest endpoint to fetch data.
the u-service architecture in itself introduces lots of latency, so the standard approach to implement front-end for u-services is to directly fetch data in browser via rest endpoint call.
even if we implement gRPC, the latency will still be there.
given this case, is adding react support in full-stack all hilla can offer? cuz this approch causes SEO issues as application first page is not SSR.
a good approach would be to make hilla just like qwik. this will counter the SEO issues by providing both SSR and CSR.
